src/templates.ts

```diff
@@ -1,74 +1,129 @@
 import { dirname, join } from 'pathe'
 import { useNuxt, addTemplate, addTypeTemplate } from '@nuxt/kit'
-import { isJSObject, NON_ALPHANUMERIC_RE } from './utils'
-import type { ExposeConfig, ResolvedTwConfig, TWConfig } from './types'
+import type { ResolvedNuxtTemplate } from 'nuxt/schema'
+import type { ExposeConfig, TWConfig } from './types'
+import resolveConfig from 'tailwindcss/resolveConfig.js'
+
+const NON_ALPHANUMERIC_RE = /^[0-9a-z]+$/i
+const isJSObject = (value: any) => typeof value === 'object' && !Array.isArray(value)
 
 /**
  * Creates MJS exports for properties of the config
  *
  * @param resolvedConfig tailwind config
  * @param maxLevel maximum level of depth
  * @param nuxt nuxt app
+ *
+ * @returns array of templates
  */
-export default function createTemplates (resolvedConfig: Partial<TWConfig> | ResolvedTwConfig, config: ExposeConfig, nuxt = useNuxt()) {
-  const dtsContent: Array<string> = []
+export default function createConfigTemplates(twConfig: Awaited<ReturnType<typeof import('./config')['default']>>, config: ExposeConfig, nuxt = useNuxt()) {
+  const templates: ResolvedNuxtTemplate<any>[] = []
+  const getTWConfig = (objPath: string[] = []) =>
+    import(twConfig.dst).then((config: TWConfig) => resolveConfig(config))
```

Comment:
had to use `import` here instead of `loadConfig` from tailwind as there were a couple of caveats; same here as well, but seems to work better with our HMR.